jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -320,6 +320,24 @@ default Resource newResource(String resource)
         return newResource(URIUtil.toURI(resource));
     }
 
+    /**
+     * Construct a resource from a string.
+     *
+     * @param resource A URL or filename.
+     * @param base Resource base.
+     * @return A Resource object, or null if the string points to a location
+     *    that does not exist
+     * @throws IllegalArgumentException if resource is invalid
+     */
+    default Resource newResource(Resource base, String resource)
+    {
+        if (StringUtil.isBlank(resource))
+            throw new IllegalArgumentException("Resource String is invalid: " + resource);
+        if (URIUtil.isRelative(resource))
+            return base.resolve(resource);
+        return newResource(URIUtil.toURI(resource));
+    }
```

Comment:
I'm wondering if instead of having a new method like this, perhaps we should be able to create a ResourceFactory with a specific base.  So the ResourceFactory associated with a context would have the contexts resource base for all methods.     The other ResourceFactories would have current working directory as their base unless otherwise configured.

Reply:
Looking more, this is definitely the wrong approach.  We already have `ResourceFactory.of(Resource baseResource)`, we just need to make sure we are using that and that we don't resolve relative strings to absolute file URIs before calling the resource factory.   I'm working on this now.

Reply:
We already have `ResourceFactory.of(Resource)` for this purpose. This looks like what we want to do here.